### PR TITLE
chore: daje as cli

### LINKED ISSUE #10: Startupping the Go CLI for Daje
Startup the Go project for the new Daje CLI

### LINKED ISSUE #11: Set up commit lint
We want to set up commit lint to verify the commit messages are conformed with a standard!

### LINKED ISSUE #12: Set up the golangci
Set up the Golang CI GitHub Action

### PR DESCRIPTION
**Pull request by davideimola**
closes #12 
closes #11 
closes #10

### CODE REVIEW

File: pkg/cmd/root/root.go [hunk 1]
```diff
@@ -0,0 +1,23 @@
+package root
+
+import (
+	"github.com/spf13/cobra"
+
+	initCmd "github.com/Schrodinger-Hat/Daje/pkg/cmd/init"
```
Comment (Wabri) [inline pkg/cmd/root/root.go:6]: Why did you use the remote path for the package init and not the local relative path?

File: pkg/cmd/init/init.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
+package init
+
+import "github.com/spf13/cobra"
+
+func NewCmdInit() *cobra.Command {
```
Comment (Wabri) [inline pkg/cmd/init/init.go:11]: Why did you use the prefix New for the function?

File: main.go [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package main
+
+import (
+	"os"
+
+	"github.com/Schrodinger-Hat/Daje/pkg/cmd/root"
```
Comment (Wabri) [inline main.go:6]: Why did you use the remote path for the package init and not the local relative path?

File: .golangci.yaml [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    suggest-new: true
+  goimports:
+    local-prefixes: github.com/Schrodinger-Hat/Daje
+  dupl:
+    threshold: 100
+  goconst:
+    min-len: 2
+    min-occurrences: 2
+linters:
+  disable-all: true
```
Comment (Wabri) [inline .golangci.yaml:16]: Is it a best practice to disable all linters and then enable them all?

File: pkg/cmd/root/root.go (hunk 1 above)
Comment (davideimola) [reply pkg/cmd/root/root.go:6]: It takes the module name put inside the `go.mod` file

https://github.com/Schrodinger-Hat/Daje/blob/59c61b8c2c5142478e27fc318c9de019815081f1/go.mod#L1

It is a good practice use something like that, let me show a few well-known examples:

https://github.com/gohugoio/hugo/blob/5b5dcb8d5a4669e1768951b452cfd53c6b25825c/go.mod#L1

https://github.com/cli/cli/blob/6783509d3bfdf55e1007dbedf34f2bdcd9fd1ec1/go.mod#L1

File: pkg/cmd/init/init.go (hunk 2 above)
Comment (davideimola) [reply pkg/cmd/init/init.go:11]: I have decided to use more or less the style used by the gh cli as you can see

https://github.com/cli/cli/blob/6783509d3bfdf55e1007dbedf34f2bdcd9fd1ec1/pkg/cmd/root/root.go#L39

File: .golangci.yaml (hunk 4 above)
Comment (davideimola) [reply .golangci.yaml:16]: I have enabled only a few of them. Not all the linters inside golangci are useful, I have found them as a good entry point for a project :smile:

**Review by Wabri — APPROVED** [review]

**Pull request merged**